vcddiff.c

```diff
@@ -989,12 +1032,22 @@ static void restore(FILE *fp, char *str, int size)
  int i;
  char *cp;
 
- ungetc(' ', fp);
+ {
+	int r=ungetc(' ', fp);
+	assert(r!=EOF);
+ }
  cp = &str[size-1];
 
- for(i =0; i< size; i++, cp--)
-	 ungetc(*cp, fp);
-
+ for(i =0; i< size; i++, cp--){
+	 int r=ungetc(*cp, fp);
+	 if(r==EOF){
+		 //because ungetc implementations are not required to support
+		 //more than one call of it in a row, this sort of usage is 
```

Comment:
```suggestion
		 //more than one call of it in a row, this sort of usage is
```

Please remove trailing spaces from your new lines.

Reply:
Note there's other places too - "git apply" of the patch file should not report any whitespace errors.